articleview.cc

```diff
@@ -364,6 +364,14 @@ ArticleView::ArticleView( QWidget * parent, ArticleNetworkAccessManager & nm, Au
            &AnkiConnector::errorText,
            this,
            [ this ]( QString const & errorText ) { emit statusBarMessage( errorText ); } );
+
+  connect( ui.definition, &ArticleWebView::loadFinished, this, [ this ]( bool ) {
+    ui.definition->show();
+
+    // search lightui+darkmode to jump to another section of this hack
+    // reset the background back to normal
+    setStyleSheet( "ArticleView {background-color: none;}" );
+  } );
```

Comment:
can this line called only once in ArticleView's  constructor .

Reply:
Sadly we cannot. The search widget is part of the ArticleView which is a QFrame.

I added a TODO to decouple them which require huge amount of work. 

![image](https://user-images.githubusercontent.com/20123683/225793045-cb5f5201-6650-41df-9c8b-9229c4b9beae.png)

Reply:
I think simply rename `ArticleView` to `ArticleWidget`  can classified the misunderstand with WebEngineView.

Reply:
![image](https://user-images.githubusercontent.com/105986/225794268-545da7a8-27eb-46a0-9d01-885759ee9765.png)
can not see the text

Reply:
Yes, the text disappeared if set the background-color of ArticleView. Some refactoring must be done to make it effcient.

![image](https://user-images.githubusercontent.com/20123683/225796883-dea5dffe-dc83-45ee-96f9-5e8c78494f17.png)

You may also think about why I don't just simply set background of the `frame` that contains the WebViewPage.

We cannot. The QFrame don't have a background. I think the usage QFrame here is useless. The QFrame should be just converted to a plain QWidget + a QVBoxlayout. It will be more efficeint and elimate this ahck.

Reply:
I prefer to convert ArticleView.ui to plain code. The code generated from ui files are not optimal (check the ui_articleview.h in the build output dir).

Reply:
auto generated code will make development easier,I do not think write from start is a good idea.

from the screenshot above , I can conclude that the articleview is not has a better name.  
It is a container contains  all these controls ,even without articleview ,we will have other container to hold all these.

Reply:
> We cannot. The QFrame don't have a background

setAttribute( Qt::WA_TranslucentBackground );

Reply:
articleview.ui only has a few widgets, and it matters for the latency between searches (which is our core feature). Making it faster doesn't hurt.

> I can conclude that the articleview is not has a better name.

I perfer to change the name into ArticleWidget (which includes a webview and other things)

Reply:
rename can be in a seperate PR

Reply:
> auto generated code will make development easier

I think it makes things harder.

For example, the code is wrong already,

In the ui file, the type of ArticleView is QWidget

https://github.com/xiaoyifang/goldendict/blob/fb6d4f9ccd439fbec6a57d3d8d7a4583d6951251/articleview.ui#L4

In the .cpp file, the type of ArticleView is QFrame

https://github.com/xiaoyifang/goldendict/blob/fb6d4f9ccd439fbec6a57d3d8d7a4583d6951251/articleview.hh#L31

For every new tab opened, there is a lot of useless dark magic involved :)